src/pages/LetterWritePage/components/LetterReceiverContainer.tsx

```diff
@@ -24,35 +25,56 @@ const ReceiverContainer = ({ onClick, isOpen }: ReceiverContainerProps) => {
   const gender = watch('gender') as '' | EqualGender;
   const worryType = watch('worryType') as '' | Worry;
 
+  const [tags, setTags] = useState<string[]>([]);
+
+  useEffect(() => {
+    const updateTags = () => {
+      console.log('하이');
+
+      const newTags = [];
+
+      if (age.length !== 0) {
+        const ageLabel = `${age[0]}~${age[1]}`;
+        newTags.push(ageLabel);
+      }
+
+      if (gender !== '') {
+        const genderLabel =
+          EQUAL_GENDER_DICT[0] === gender ? '모두에게' : '동성에게';
+        newTags.push(genderLabel);
+      }
+
+      if (worryType !== '') {
+        const worryLabel = WORRY_DICT[worryType];
+        newTags.push(worryLabel);
+      }
+
+      setTags(newTags);
+    };
+
+    updateTags();
+  }, [age, gender, worryType]);
```

Comment:
p3;
> 외부 시스템이 관여하지 않는 경우 (예를 들어 일부 props 또는 state가 변경될 때 컴포넌트의 state를 업데이트하려는 경우), Effect가 필요하지 않습니다.
https://ko.react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes

useEffect는 **리액트 외부 영역과의 동기화가 필요**할 때 주로 사용되는 훅이라는 생각이 들어요! 
또한 태그를 상태로 관리하지 않아도 `watch` 로 구독한 상태를 가져와서 사용할 수 있을 것 같아요 😊

```suggestion
  const tags = [];

  if (age.length !== 0) {
    const ageLabel = `${age[0]}~${age[1]}`;
    tags.push(ageLabel);
  }

  if (gender !== '') {
    const genderLabel =
      EQUAL_GENDER_DICT[0] === gender ? '모두에게' : '동성에게';
    tags.push(genderLabel);
  }

  if (worryType !== '') {
    const worryLabel = WORRY_DICT[worryType];
    tags.push(worryLabel);
  }
```